QPSK_UT/Demodulator_UT.py

```diff
@@ -31,38 +33,57 @@
                 1, -1, 1, 1, -1, 1, -1, -1, 1, 1, -1, -1, 1, 1, -1, -1, 1, 1, -1, 1,
                 1, -1, 1, 1, -1, 1, -1, -1, 1, 1, -1, -1, 1, 1, -1, -1, 1, 1, -1, 1]
 __OUTPUT_BITS = [1 if x > 0 else 0 for x in __INPUT_BITS]
+__PSF = cp.rrcosfilter(int(__SYMBOL_LENGTH_IN_BITS)*10 , 0.35, __SYMBOL_LENGTH_IN_BITS / __SAMPLE_RATE , __SAMPLE_RATE)[1]
 
-def __calcSignal() :
-    time = numpy.linspace(0, __NUM_OF_PERIODS_IN_SYMBOL / __CARRIER_FREQ, __SYMBOL_LENGTH_IN_BITS)
-    angle = 2 * numpy.pi * __CARRIER_FREQ * time + __FI
 
-    sig = []
-    for i in range(int(len(__INPUT_BITS) / 2)):
-        sig.extend(__INPUT_BITS[i * 2] * numpy.cos(angle) - 1j*__INPUT_BITS[i * 2 + 1] * numpy.sin(angle))
-    return sig
+########################################################################################################################
+#       FUNCTIONS
+########################################################################################################################
+
+def __filterSignal(inSig):
+    filtered = np.convolve(__PSF, inSig)
+    filtered = filtered[int(__SYMBOL_LENGTH_IN_BITS*5): -int(__SYMBOL_LENGTH_IN_BITS*5)+1]
+    return filtered
+
```

Comment:
też dodatkowa linia xD

Reply:
od calcSignal do shouldDemodulateMostOfInputBitsWithNoise powinna być 1 linii więcej przed każda def metody

Reply:
o ile przed klasami poprawię to o tym zapomnij, jak dla mnie bez sensu i nie mam zamiaru tego poprawiać